Katello/katello
# Fixes #33376 - catch & return new Pulp mirror on sync error

### What are the changes introduced in this pull request?

Pulp introduced a new error for repositories that don't work with mirror-on-sync (https://github.com/dralley/pulp_rpm/commit/d4d088ae8d2e20b6d69d8b321b57ebee2afb3163#diff-fb406f31027edc1c5b65abe7635bbd2830a7cc05c24a487344d7649017d14628L107).  This PR catches that error and makes it more friendly for Katello users.

### What are the testing steps for this pull request?
1) Create a yum repo syncing from https://dl.fedoraproject.org/pub/epel/7/x86_64/
2) Set mirror on sync to on
3) Sync the repo and see the error: "Please disable 'mirror on sync' because the upstream repository refers to external resources."
